src/pydvl/value/shapley/classwise.py

```diff
@@ -0,0 +1,216 @@
+        for each permutation.
+    :param progress: Whether to display progress bars for each job.
+    :return: ValuationResult object with the data values.
+    """
+    _check_classwise_shapley_utility(u)
+
+    result = ValuationResult.zeros(
+        algorithm="classwise_shapley",
+    )
+
+    x_train, y_train = u.data.get_training_data(update_indices)
+    unique_labels = np.unique(y_train)
+    pbar = tqdm(disable=not progress, position=0, total=100, unit="%")
+    scorer = cast(ClassWiseScorer, u.scorer)
+
+    for n_step in itertools.count():
+
+        pbar.n = 100 * done.completion()
+        pbar.refresh()
+
+        for label in unique_labels:
+            logger.debug(
+                f"Sampling subset #{n_step} complement index set of class '{label}'."
+            )
+            scorer.label = label
+            result += permutation_montecarlo_classwise_shapley(
+                u,
+                label,
+                done=MaxChecks(n_resample_complement_sets),
+                truncation=truncation,
+            )
+
+        if done(result):
+            break
+
+    if normalize_score:
+        _normalize_classwise_shapley_values(result, u)
+
+    return result
+
+
+def _check_classwise_shapley_utility(u: Utility):
+    """
+    Checks whether the passed utility object is suitable for computing the class-wise
+    shapley values. Therefore, it is checked that the data passed is a pure
+    classification dataset. If the function passes without check it is considered to be
+    run successful.
+
+    :param u: Utility object with model, data, and scoring function. The scoring
+        function has to be of type :class:`~pydvl.utils.score.ClassWiseScorer`.
+    :raises: ValueError: If ``u.data`` is not a classification problem.
+    :raises: ValueError: If ``u.scorer`` is not an instance of
+        :class:`~pydvl.utils.score.ClassWiseScorer`
+    """
+
+    dim_correct = u.data.y_train.ndim == 1 and u.data.y_test.ndim == 1
+    is_integral = all(
+        map(
+            lambda v: isinstance(v, numbers.Integral), (*u.data.y_train, *u.data.y_test)
+        )
+    )
+    if not dim_correct or not is_integral:
+        raise ValueError(
+            "The supplied dataset has to be a 1-dimensional classification dataset."
+        )
+
+    if not isinstance(u.scorer, ClassWiseScorer):
+        raise ValueError(
+            "Please set a subclass of ClassWiseScorer object as scorer object of the"
+            " utility. See"
+            " scoring argument of Utility."
+        )
+
+
+def _normalize_classwise_shapley_values(
+    result: ValuationResult,
+    u: Utility,
+) -> ValuationResult:
+    """
+    Normalize a valuation result by a factor specific to class-wise shapley. Each value
+    corresponds to a class c and gets normalized by multiplying `in-cls-score / sigma`.
+    See also footcite:t:`schoch_csshapley_2022` for more details.
```

Comment:
Please explain the terms that you introduce. What's sigma? Only refer to the paper as a last recourse. Users should have all information in the documentation